### PR TITLE
Add Panel2.cfg support

### PR DESCRIPTION
**Pull request by cwfitzgerald**
Looks quite similar to panel2, but they are different enough to not warrant combining.

Notable Changes:
- Added HexColor3 and HexColor4 types for colors coming from HTML `#XXXXXX` notation.
- Blu-tacked on file filtering to the corpus tester to reduce runtimes, will be fixed better by the PR that solves #39.
- The `Subject` value type has a custom parser using nom due to its complexity.

### CODE REVIEW

File: bve/src/datatypes.rs [hunk 1]
```diff
@@ -62,3 +64,68 @@ pub type ColorF32RG = Vector2<f32>;
 pub type ColorF32RGB = Vector3<f32>;
 /// RGBA color: 32-bit float per channel
 pub type ColorF32RGBA = Vector4<f32>;
+
+/// A color that is specifically formatted using normal HTML hex notation.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct HexColor3(pub ColorU8RGB);
```
Comment (DethRaid) [inline bve/src/datatypes.rs]: HexColorRgb

File: bve/src/datatypes.rs [hunk 2]
```diff
@@ -62,3 +64,68 @@ pub type ColorF32RG = Vector2<f32>;
...
+            (number & 0xFF) as u8,
+            ((number >> 8) & 0xFF) as u8,
+            ((number >> 16) & 0xFF) as u8,
+        )))
+    }
+}
+
+/// A color with alpha that is specifically formatted using normal HTML hex notation.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct HexColor4(pub ColorU8RGBA);
```
Comment (DethRaid) [inline bve/src/datatypes.rs]: HexColorRgba

File: bve/src/parse/panel2_cfg/sections/subject.rs [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+            variant_value!('a, "atc", Ats, 271),
+            variant_index!('a, "ats", Ats),
+            variant!('a, "locobrakecylinder", LocoBrakeCylinder),
+            variant!('a, "bc", BrakeCylinder),
+            variant!('a, "locobrakepipe", LocoBrakePipe),
+            variant!('a, "bp", BrakePipe),
+            variant!('a, "brake", Brake),
+            variant!('a, "locobrake", LocoBrake),
+            variant!('a, "csc", ConstSpeedSystem),
+            // These 4 must come before door, as door is a prefix for them and will partially parse, then fail
```
Comment (DethRaid) [inline bve/src/parse/panel2_cfg/sections/subject.rs]: 10/10 file format vv useful

File: bve/src/parse/panel2_cfg/sections/this.rs [hunk 4]
```diff
@@ -0,0 +1,35 @@
+use crate::HexColor3;
+use bve_derive::FromKVPSection;
+use cgmath::{Array, Vector2};
+
+#[derive(Debug, Clone, PartialEq, FromKVPSection)]
+pub struct ThisSection {
```
Comment (DethRaid) [inline bve/src/parse/panel2_cfg/sections/this.rs]: I don't like this name but I guess it comes directly from BVE itself :<

Comment (cwfitzgerald) [reply bve/src/parse/panel2_cfg/sections/this.rs]: Yup, it's a section called `this`

File: bve/src/datatypes.rs (hunk 1 above)
Comment (cwfitzgerald) [reply bve/src/datatypes.rs]: Fiiiiiiiiiine

File: bve/src/parse/panel2_cfg/sections/subject.rs (hunk 3 above)
Comment (cwfitzgerald) [reply bve/src/parse/panel2_cfg/sections/subject.rs]: The file format is fine, just how recursive decent parsers work

Comment (DethRaid) [reply bve/src/parse/panel2_cfg/sections/subject.rs]: Nope I blame the file format and you can't stop me

**Pull request merged**